src/transformers/models/layoutlm/configuration_layoutlm.py

```diff
@@ -125,3 +131,84 @@ def __init__(
             **kwargs,
         )
         self.max_2d_position_embeddings = max_2d_position_embeddings
+
+
+class LayoutLMOnnxConfig(OnnxConfig):
+
+    DEFAULT_MAX_2D_POSITION = 1000
+
+    @property
+    def inputs(self) -> Mapping[str, Mapping[int, str]]:
+        return OrderedDict(
+            [
+                ("input_ids", {0: "batch", 1: "sequence"}),
+                ("bbox", {0: "batch", 1: "sequence"}),
```

Comment:
Oh how did you know that? Is this mentioned in the docs?

Reply:
Its not explicitly mentioned in the docs. But the tutorial link you sent shows the config for Bert, and that does not specify the static dimension in the output (the 768 dimension in "last_hidden_state" and "pooler_output"). So I went through the code in src/transformers/onnx to verify this. The dimensions returned by the config file are passed on to the export function provided by PyTorch in the torch.onnx module as shown in src/transformers/onxx/conver.py:

```
    export(
        model,
        (model_inputs,),
        f=output.as_posix(),
        input_names=list(config.inputs.keys()),
        output_names=onnx_outputs,
        dynamic_axes={name: axes for name, axes in chain(config.inputs.items(), config.outputs.items())},
        do_constant_folding=True,
        use_external_data_format=config.use_external_data_format(model.num_parameters()),
        enable_onnx_checker=True,
        opset_version=opset,
    )

```
As you can see, the shapes provided by the config file are passed to the dynamic_axes parameter in the function call.

I also verified the change by converting a LayoutLM model to ONNX- the verification step that runs during conversion confirmed that the outputs of the converted model are similar to the original. Visualizing the converted ONNX model using Netron also shows that the input and output sizes are as follows:

```
Inputs:     
------     
name: input_ids    
type: int64[batch,sequence]    
    
name: bbox    
type: int64[batch,sequence,4]    
    
name: attention_mask    
type: int64[batch,sequence]    
    
name: token_type_ids    
type: int64[batch,sequence]    
    
Outputs:     
-------     
name: last_hidden_state    
type: float32[batch,sequence,768]    
    
name: pooler_output    
type: float32[batch,768]
```

I do think that this should be explicitly mentioned in the docs though, so that it will be easier for anyone trying to convert custom models.

Reply:
@nishprabhu you're totally right, this is about mapping the dynamic axis for each tensor (_i.e. axis we don't know beforehand or can change between two successive calls_).

In this case, if the bounding box dimensions are constant (x1, y2, x2, y2) then ONNX will infer and reports this in the traced model 👍🏻

Reply:
We will update the documentation to better reflect this 👍🏻

Reply:
That's actually interesting to know, I've never worked with ONNX, but this seems indeed pretty important.